src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1693,7 +1693,16 @@ void TorrentHandle::handleTrackerErrorAlert(const lt::tracker_error_alert *p)
 
     m_trackerInfos[trackerUrl].lastMessage = message;
 
-    m_session->handleTorrentTrackerError(this, trackerUrl);
+    // Starting with libtorrent 1.2.x each tracker has multiple local endpoints from which
+    // an announce is attempted. Some endpoints might succeed while others might fail.
+    // Emit the signal only if all endpoints have failed. TrackerEntry::isWorking() returns
+    // true if at least one endpoint works.
+    const QVector<TrackerEntry> trackerList = trackers();
+    const auto iter = std::find_if(trackerList.cbegin(), trackerList.cend(), [&trackerUrl](const TrackerEntry &entry) {
```

Comment:
Brace should be on its own line (like in regular functions).

Reply:
fixed